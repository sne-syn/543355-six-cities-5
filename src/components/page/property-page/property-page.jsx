import React from 'react';
import PropTypes from 'prop-types';
import Page from '../page';
import Header from '../../header/header';
import FavoriteButton from '../../favorite-button/favorite-button';
import Host from '../../host/host';
import MapSection from '../../map-section/map-section';
import ReviewsList from '../../reviews-list/reviews-list';
import NearPlaces from '../../near-places/near-places';
import StarBar from '../../star-bar/star-bar';
import {CITIES, AccomodationTypes, CountCards} from '../../../utils/const';
import {capitalizeChar} from '../../../utils/common';

const PropertyPage = ({offer, reviews, hosts, isLogged, offers}) => {
  const nearPlacesToRender = offers.slice(0, CountCards.NEAR_LIST);
  return (
    <Page pageClass={`page`}>
      <Header isLogged={isLogged}/>
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {offer.images.slice(0, 6).map((img, i) => (
                <div key={i} className="property__image-wrapper">
                  <img className="property__image" src={img} alt="Photo studio" />
                </div>
              ))}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {offer.isPremium && (
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              )}
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {offer.title}
                </h1>
                <FavoriteButton isFavorite={offer.isFavorite} componentName={`property`} />
              </div>
              <StarBar rating={offer.rating} containerClassName={`property`}>
                <span className="property__rating-value rating__value">{offer.rating.toFixed(1)}</span>
              </ StarBar>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {AccomodationTypes[offer.type]}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {offer.bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                Max {offer.maxGuests} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{offer.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {offer.features.map((feature, i) => (
                    <li key={i} className="property__inside-item">
                      {capitalizeChar(feature)}
                    </li>
                  ))}
                </ul>
              </div>

              <Host description={offer.description} hostID={offer.host} hosts={hosts}/>
              <ReviewsList reviews={reviews} isLogged={isLogged}/>

            </div>
          </div>
          <section className="property__map map">
            <MapSection currentCity={offer.city} offersToRender={offer}/>
          </section>
        </section>
        <div className="container">
          <NearPlaces offers={nearPlacesToRender}/>
        </div>
      </main>
    </Page>
  );
};

PropertyPage.propTypes = {
  isLogged: PropTypes.bool.isRequired,
  offers: PropTypes.array.isRequired,
  offer: PropTypes.shape({
    id: PropTypes.string.isRequired,
    city: PropTypes.oneOf([...CITIES]).isRequired,
    title: PropTypes.string.isRequired,
    images: PropTypes.array.isRequired,
    price: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    features: PropTypes.array.isRequired,
    bedrooms: PropTypes.number.isRequired,
    maxGuests: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    reviews: PropTypes.array.isRequired,
    host: PropTypes.string.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  }).isRequired,
  hosts: PropTypes.array.isRequired,
  reviews: PropTypes.array.isRequired,
};

export default PropertyPage;
