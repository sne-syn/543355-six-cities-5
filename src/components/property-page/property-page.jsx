import React from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header';
import PremiumMark from '../premium-mark/premium-mark';
import FavoriteButton from '../favorite-button/favorite-button';
import Host from '../host/host';
import MapSection from '../map-section/map-section';
import PropertyFeatures from '../property-features/property-features';
import ReviewsList from '../reviews-list/reviews-list';
import NearPlaces from '../near-places/near-places';
import StarBar from '../star-bar/star-bar';
import {capitalizeChar} from '../../utils/common';
const nearPlacesCount = 3;

const PropertyPage = (props) => {
  const {offer, offers} = props;
  const nearPlacesToRender = offers.slice(0, nearPlacesCount);
  return (
    <div className="page">
      <Header {...props}/>
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {offer.images.slice(0, 6).map((img, i) => (
                <div key={i} className="property__image-wrapper">
                  <img className="property__image" src={img} alt="Photo studio" />
                </div>
              ))}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {offer.isPremium && (<PremiumMark componentName={`property`}/>)}
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {offer.title}
                </h1>
                <FavoriteButton defaultOnValue={offer.isFavorite} componentName={`property`} />
              </div>
              <StarBar rating={offer.rating} containerClassName={`property`}>
                <span className="property__rating-value rating__value">{offer.rating.toFixed(1)}</span>
              </ StarBar>
              <PropertyFeatures type={offer.type} bedrooms={offer.bedrooms} maxGuests={offer.maxGuests} />
              <div className="property__price">
                <b className="property__price-value">&euro;{offer.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {offer.features.map((feature, i) => (
                    <li key={i} className="property__inside-item">
                      {capitalizeChar(feature)}
                    </li>
                  ))}
                </ul>
              </div>

              <Host description={offer.description} hostID={offer.host} {...props}/>
              <ReviewsList {...props}/>
            </div>
          </div>
          <section className="property__map map">
            <MapSection activeElement={offer.city.name} offersToRender={nearPlacesToRender} activeOffer={offer}/>
          </section>
        </section>
        <div className="container">
          <NearPlaces offers={nearPlacesToRender}/>
        </div>
      </main>
    </div>
  );
};

PropertyPage.propTypes = {
  offers: PropTypes.array.isRequired,
  offer: PropTypes.shape({
    id: PropTypes.string.isRequired,
    city: PropTypes.shape({
      location: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
        zoom: PropTypes.number.isRequired,
      }).isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    title: PropTypes.string.isRequired,
    images: PropTypes.array.isRequired,
    price: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    features: PropTypes.array.isRequired,
    bedrooms: PropTypes.number.isRequired,
    maxGuests: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    reviews: PropTypes.array.isRequired,
    host: PropTypes.string.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  }).isRequired,
};
export default PropertyPage;
