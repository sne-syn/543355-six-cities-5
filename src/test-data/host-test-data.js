export const hostPro = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: true,
  name: `Janette`,
};
export const host = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: false,
  name: `Janette`,
};
