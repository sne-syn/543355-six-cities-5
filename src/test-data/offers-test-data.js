export const offers = [{
  bedrooms: 1,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Paris`,
  },
  description: `Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish`,
  goods: [`cabel TV`, `heating`, `kitchen`, `towels`, `washing machine`, `washer`, `dishwasher`, `conditioner`, `elevator`, `coffee machine`],
  host: {
    avatar: `https://robohash.org/28?set=set2&size=74x74`,
    id: 4,
    isPro: false,
    name: `Salom`,
  },
  id: 1,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/34LWdhj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/33Qpecj`, `https://bit.ly/30YVQif`, `https://bit.ly/36TgIv7`, `https://bit.ly/3lEt2DG`],
  isFavorite: true,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 8,
  price: 207,
  rating: 1.4018800815887484,
  title: `La Latina- Small Charming Studio`,
  type: `room`,
}, {
  bedrooms: 3,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Brussels`,
  },
  description: `In full nature and very close to public transport. Green area surrounded by all kinds of sports facilities including a golf course, paddle, artificial grass soccer, pelota court. In an urbanization of villas with landscaped plots of about 500 m2`,
  goods: [`elevator`, `dishwasher`, `coffee machine`, `washer`, `washing machine`],
  host: {
    avatar: `https://robohash.org/82?set=set2&size=74x74`,
    id: 6,
    isPro: false,
    name: `Whitney`,
  },
  id: 2,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/30YVQif`, `https://bit.ly/3lEt2DG`, `https://bit.ly/36TgIv7`, `https://bit.ly/33Qpecj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/34LWdhj`],
  isFavorite: false,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 5,
  price: 194,
  rating: 1.9069547983791004,
  title: `Estudio/suite completo en el centro de Segovia`,
  type: `house`,
}];
