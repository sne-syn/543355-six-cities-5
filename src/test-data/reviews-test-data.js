export const review = {
  author: `Beaulah`,
  avatar: `https://robohash.org/54?set=set2&size=54x54`,
  date: `2004-09-06T16:12:32.554Z`,
  id: 0,
  rating: 1,
  comment: `Antoni is a great host and us a wonderful home. His suggestions for places to go were killer!10/10 would stay again.`,
};

export const reviews = [{
  author: `Ione`,
  avatar: `https://robohash.org/98?set=set2&size=54x54`,
  date: `2007-10-13T16:12:32.554Z`,
  id: 1,
  rating: 4,
  comment: `There’s a stunning beach with great snorkelling just a 6min drive, a few other beautiful beaches close by too.`,
}, {
  author: `Bruno`,
  avatar: `https://robohash.org/53?set=set2&size=54x54`,
  date: `2005-08-25T16:12:32.554Z`,
  id: 2,
  rating: 3,
  comment: `Maria’s place is at a perfect location, very easy to get around with public transportation and walking distance to awesome bars and restaurants. Her place is great, very clean and has everything you need.`,
}];
