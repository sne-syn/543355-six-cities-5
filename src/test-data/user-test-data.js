export const user = {
  userAvatar: `https://robohash.org/82?set=set2&size=74x74`,
  userEmail: `trust_no_1@gmail.com`,
  userId: 1,
  userIsPro: false,
  userName: `fox_mulder`
};
