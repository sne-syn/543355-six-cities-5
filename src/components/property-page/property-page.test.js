import React from 'react';
import renderer from 'react-test-renderer';
import {PropertyPage} from './property-page';
import {offer, offerPremium} from '../../test-data/offer-test-data';

const noop = () => {};
jest.mock(`../favorite-button/favorite-button`, () => `FavoriteButton`);
jest.mock(`../header/header`, () => `Header`);
jest.mock(`../host/host`, () => `Host`);
jest.mock(`../loader/loader`, () => `Loader`);
jest.mock(`../map-section/map-section`, () => `MapSection`);
jest.mock(`../near-places/near-places`, () => `NearPlaces`);
jest.mock(`../premium-mark/premium-mark`, () => `PremiumMark`);
jest.mock(`../property-features/property-features`, () => `PropertyFeatures`);
jest.mock(`../reviews-list/reviews-list`, () => `ReviewsList`);
jest.mock(`../star-bar/star-bar`, () => `StarBar`);

describe(`<PropertyPage /> test`, () => {
  it(`renders PropertyPage with loader`, () => {
    const tree = renderer
    .create(
        <PropertyPage
          authorizationStatus={`AUTH`}
          fetchPropertyPageAction={noop}
          id={`24`}
          loading={true}
          nearPlaces={[]}
          offer={{}}
          reviews={[]}
        />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders PropertyPage with fetched data`, () => {
    const tree = renderer
    .create(
        <PropertyPage
          authorizationStatus={`AUTH`}
          fetchPropertyPageAction={noop}
          id={`24`}
          loading={false}
          nearPlaces={[]}
          offer={offer}
          reviews={[]}
        />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders PropertyPage with fetched data and Premium mark`, () => {
    const tree = renderer
    .create(
        <PropertyPage
          authorizationStatus={`AUTH`}
          fetchPropertyPageAction={noop}
          id={`24`}
          loading={false}
          nearPlaces={[]}
          offer={offerPremium}
          reviews={[]}
        />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
