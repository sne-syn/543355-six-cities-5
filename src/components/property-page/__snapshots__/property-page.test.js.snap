// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertyPage /> test renders PropertyPage with fetched data 1`] = `
<div
  className="page"
>
  <Header />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/34LWdhj"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/36Ys6Wn"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/33Qpecj"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/30YVQif"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/36TgIv7"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/3lEt2DG"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              La Latina- Small Charming Studio
            </h1>
            <FavoriteButton
              componentName="property"
              defaultOnValue={false}
              offerId={1}
            />
          </div>
          <StarBar
            containerClassName="property"
            rating={1.4018800815887484}
          >
            <span
              className="property__rating-value rating__value"
            >
              1.4
            </span>
          </StarBar>
          <PropertyFeatures
            bedrooms={1}
            maxGuests={8}
            type="room"
          />
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              207
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Cabel TV
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Conditioner
              </li>
              <li
                className="property__inside-item"
              >
                Elevator
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
            </ul>
          </div>
          <Host
            authorizationStatus="AUTH"
            description="Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish"
            fetchPropertyPageAction={[Function]}
            host={
              Object {
                "avatar": "https://robohash.org/28?set=set2&size=74x74",
                "id": 4,
                "isPro": false,
                "name": "Salom",
              }
            }
            id="24"
            loading={false}
            nearPlaces={Array []}
            offer={
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Paris",
                },
                "description": "Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish",
                "goods": Array [
                  "cabel TV",
                  "heating",
                  "kitchen",
                  "towels",
                  "washing machine",
                  "washer",
                  "dishwasher",
                  "conditioner",
                  "elevator",
                  "coffee machine",
                ],
                "host": Object {
                  "avatar": "https://robohash.org/28?set=set2&size=74x74",
                  "id": 4,
                  "isPro": false,
                  "name": "Salom",
                },
                "id": 1,
                "images": Array [
                  "https://bit.ly/34LWdhj",
                  "https://bit.ly/36Ys6Wn",
                  "https://bit.ly/33Qpecj",
                  "https://bit.ly/30YVQif",
                  "https://bit.ly/36TgIv7",
                  "https://bit.ly/3lEt2DG",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 48.865610000000004,
                  "longitude": 2.350499,
                  "zoom": 5,
                },
                "maxGuests": 8,
                "previewImage": "https://bit.ly/34LWdhj",
                "price": 207,
                "rating": 1.4018800815887484,
                "title": "La Latina- Small Charming Studio",
                "type": "room",
              }
            }
            reviews={Array []}
          />
          <ReviewsList
            authorizationStatus="AUTH"
            reviews={Array []}
          />
        </div>
      </div>
      <section
        className="property__map map"
      >
        <MapSection
          activeCity="Paris"
          activeOffer={1}
          offersToShowOnMap={
            Array [
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Paris",
                },
                "description": "Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish",
                "goods": Array [
                  "cabel TV",
                  "heating",
                  "kitchen",
                  "towels",
                  "washing machine",
                  "washer",
                  "dishwasher",
                  "conditioner",
                  "elevator",
                  "coffee machine",
                ],
                "host": Object {
                  "avatar": "https://robohash.org/28?set=set2&size=74x74",
                  "id": 4,
                  "isPro": false,
                  "name": "Salom",
                },
                "id": 1,
                "images": Array [
                  "https://bit.ly/34LWdhj",
                  "https://bit.ly/36Ys6Wn",
                  "https://bit.ly/33Qpecj",
                  "https://bit.ly/30YVQif",
                  "https://bit.ly/36TgIv7",
                  "https://bit.ly/3lEt2DG",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 48.865610000000004,
                  "longitude": 2.350499,
                  "zoom": 5,
                },
                "maxGuests": 8,
                "previewImage": "https://bit.ly/34LWdhj",
                "price": 207,
                "rating": 1.4018800815887484,
                "title": "La Latina- Small Charming Studio",
                "type": "room",
              },
            ]
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <NearPlaces
        offers={Array []}
      />
    </div>
  </main>
</div>
`;

exports[`<PropertyPage /> test renders PropertyPage with fetched data and Premium mark 1`] = `
<div
  className="page"
>
  <Header />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/34LWdhj"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/36Ys6Wn"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/33Qpecj"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/30YVQif"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/36TgIv7"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://bit.ly/3lEt2DG"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <PremiumMark
            componentName="property"
          />
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              La Latina - Small Charming Studio
            </h1>
            <FavoriteButton
              componentName="property"
              defaultOnValue={true}
              offerId={1}
            />
          </div>
          <StarBar
            containerClassName="property"
            rating={1.4018800815887484}
          >
            <span
              className="property__rating-value rating__value"
            >
              1.4
            </span>
          </StarBar>
          <PropertyFeatures
            bedrooms={1}
            maxGuests={8}
            type="room"
          />
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              207
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Cabel TV
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Conditioner
              </li>
              <li
                className="property__inside-item"
              >
                Elevator
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
            </ul>
          </div>
          <Host
            authorizationStatus="AUTH"
            description="Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish"
            fetchPropertyPageAction={[Function]}
            host={
              Object {
                "avatar": "https://robohash.org/28?set=set2&size=74x74",
                "id": 4,
                "isPro": false,
                "name": "Salom",
              }
            }
            id="24"
            loading={false}
            nearPlaces={Array []}
            offer={
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Paris",
                },
                "description": "Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish",
                "goods": Array [
                  "cabel TV",
                  "heating",
                  "kitchen",
                  "towels",
                  "washing machine",
                  "washer",
                  "dishwasher",
                  "conditioner",
                  "elevator",
                  "coffee machine",
                ],
                "host": Object {
                  "avatar": "https://robohash.org/28?set=set2&size=74x74",
                  "id": 4,
                  "isPro": false,
                  "name": "Salom",
                },
                "id": 1,
                "images": Array [
                  "https://bit.ly/34LWdhj",
                  "https://bit.ly/36Ys6Wn",
                  "https://bit.ly/33Qpecj",
                  "https://bit.ly/30YVQif",
                  "https://bit.ly/36TgIv7",
                  "https://bit.ly/3lEt2DG",
                ],
                "isFavorite": true,
                "isPremium": true,
                "location": Object {
                  "latitude": 48.865610000000004,
                  "longitude": 2.350499,
                  "zoom": 5,
                },
                "maxGuests": 8,
                "previewImage": "https://bit.ly/34LWdhj",
                "price": 207,
                "rating": 1.4018800815887484,
                "title": "La Latina - Small Charming Studio",
                "type": "room",
              }
            }
            reviews={Array []}
          />
          <ReviewsList
            authorizationStatus="AUTH"
            reviews={Array []}
          />
        </div>
      </div>
      <section
        className="property__map map"
      >
        <MapSection
          activeCity="Paris"
          activeOffer={1}
          offersToShowOnMap={
            Array [
              Object {
                "bedrooms": 1,
                "city": Object {
                  "location": Object {
                    "latitude": 52.370216,
                    "longitude": 4.895168,
                    "zoom": 10,
                  },
                  "name": "Paris",
                },
                "description": "Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish",
                "goods": Array [
                  "cabel TV",
                  "heating",
                  "kitchen",
                  "towels",
                  "washing machine",
                  "washer",
                  "dishwasher",
                  "conditioner",
                  "elevator",
                  "coffee machine",
                ],
                "host": Object {
                  "avatar": "https://robohash.org/28?set=set2&size=74x74",
                  "id": 4,
                  "isPro": false,
                  "name": "Salom",
                },
                "id": 1,
                "images": Array [
                  "https://bit.ly/34LWdhj",
                  "https://bit.ly/36Ys6Wn",
                  "https://bit.ly/33Qpecj",
                  "https://bit.ly/30YVQif",
                  "https://bit.ly/36TgIv7",
                  "https://bit.ly/3lEt2DG",
                ],
                "isFavorite": true,
                "isPremium": true,
                "location": Object {
                  "latitude": 48.865610000000004,
                  "longitude": 2.350499,
                  "zoom": 5,
                },
                "maxGuests": 8,
                "previewImage": "https://bit.ly/34LWdhj",
                "price": 207,
                "rating": 1.4018800815887484,
                "title": "La Latina - Small Charming Studio",
                "type": "room",
              },
            ]
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <NearPlaces
        offers={Array []}
      />
    </div>
  </main>
</div>
`;

exports[`<PropertyPage /> test renders PropertyPage with loader 1`] = `
<div
  className="page"
>
  <Header />
  <Loader />
</div>
`;
